test/celeritas/field/FieldPropagator.test.cc

```diff
@@ -574,6 +574,76 @@ TEST_F(TwoBoxTest, electron_corner_hit)
     }
 }
 
+// Endpoint of a step is very close to the boundary.
+TEST_F(TwoBoxTest, electron_step_endpoint)
+{
+    auto particle = this->init_particle(
+        this->particle()->find(pdg::electron()), MevEnergy{10});
+    UniformZField      field(unit_radius_field_strength);
+    FieldDriverOptions driver_options;
+    driver_options.delta_intersection = 0.1;
```

Comment:
0.1*units::centimeter?  Does this test also work with the smaller value of delta_intersection?

Reply:
Yes; I changed it to an absurdly large value for testing purposes (it's easier to see the difference in the high digits, and we can be sure that roundoff errors and tolerances don't get mixed up).